src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -30,6 +31,8 @@ public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTyp
             {
                 NullableAnnotation: NullableAnnotation.Annotated
             };
+            IsParsable = TryGetParsability(parameter, wellKnownTypes, out var parsingBlockEmitter);
```

Comment:
When this PR and https://github.com/dotnet/aspnetcore/pull/46715 are consolidated, we'll have to set this parsability state in multiple branches.

The `RequestDelegateFactory` currently implements the check for the `TryParse` method at the top-level instead of within each scenario:

https://github.com/dotnet/aspnetcore/blob/86e3a4bc2d85794b6470539ddd29b9d3e194f546/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L796

However, I think this approach is fine because we want to separate out the logic for producing the assigning code from the parsing logic.